ksql-engine/src/main/java/io/confluent/ksql/function/udaf/max/IntegerMaxKudaf.java

```diff
@@ -0,0 +1,61 @@
+/**
+ * Copyright 2017 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+package io.confluent.ksql.function.udaf.max;
+
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.streams.kstream.Merger;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+import io.confluent.ksql.function.KsqlAggregateFunction;
+import io.confluent.ksql.parser.tree.Expression;
+
+public class IntegerMaxKudaf extends KsqlAggregateFunction<Integer, Integer> {
+
+  public IntegerMaxKudaf(Integer argIndexInValue) {
+    super(argIndexInValue, () -> Integer.MIN_VALUE, Schema.INT32_SCHEMA,
+          Arrays.asList(Schema.INT32_SCHEMA)
+    );
+  }
+
+  @Override
+  public Integer aggregate(Integer currentVal, Integer currentAggVal) {
+    if (currentVal > currentAggVal) {
+      return currentVal;
+    }
+    return currentAggVal;
+  }
+
+  @Override
+  public Merger<String, Integer> getMerger() {
+    return (aggKey, aggOne, aggTwo) -> {
+      if (aggOne > aggTwo) {
+        return aggOne;
+      }
+      return aggTwo;
+    };
+  }
+
+  @Override
+  public KsqlAggregateFunction<Integer, Integer> getInstance(Map<String, Integer> expressionNames,
```

Comment:
Side note: this seems like a really strange patten. It looks like this class being two things: both a UDF and a Factory.

Reply:
Yes, we can have a UDF Factory instead of this. This needs to be done for all the functions. Created an issue for it: https://github.com/confluentinc/ksql/issues/864